packages/eslint-plugin/tests/rules/no-unnecessary-type-parameters.test.ts

```diff
@@ -0,0 +1,496 @@
+    'type Fn = <T extends string>(input: T) => `a${T}b`;',
+    'type Fn = new <T>(input: T) => T;',
+    'type Fn = <T>(input: Partial<T>) => input is T;',
+    'type Fn = <T>(input: T) => { [K in keyof T]: K };',
+    'type Fn = <T>(input: T) => { [K in keyof T as K]: string };',
+    'type Fn = <T>(input: T) => { [K in keyof T as `${K & string}`]: string };',
+    'type Fn = <T>(input: T) => Partial<T>;',
+    'type Fn = <T>(input: { [i: number]: T }) => T;',
+    'type Fn = <T>(input: { [i: number]: T }) => Partial<T>;',
+    'type Fn = <T>(input: { [i: string]: T }) => Partial<T>;',
+    'type Fn = <T>(input: T) => { [i: number]: T };',
+    'type Fn = <T>(input: T) => { [i: string]: T };',
+    "type Fn = <T extends unknown[]>(input: T) => Omit<T, 'length'>;",
+    `
+      import type { AST_NODE_TYPES, TSESTree } from '@typescript-eslint/types';
+
+      declare const isNodeOfType: <NodeType extends AST_NODE_TYPES>(
+        nodeType: NodeType,
+      ) => node is Extract<TSESTree.Node, { type: NodeType }>;
+    `,
+    `
+      import type { AST_NODE_TYPES, TSESTree } from '@typescript-eslint/types';
+
+      const isNodeOfType =
+        <NodeType extends AST_NODE_TYPES>(nodeType: NodeType) =>
+        (
+          node: TSESTree.Node | null,
+        ): node is Extract<TSESTree.Node, { type: NodeType }> =>
+          node?.type === nodeType;
+    `,
+    `
+      interface I {
+        <T>(value: T): T;
+      }
+    `,
+    `
+      interface I {
+        new <T>(value: T): T;
+      }
+    `,
+    `
+      function identity<T>(arg: T): T {
+        return arg;
+      }
+    `,
+    `
+      function printProperty<T>(obj: T, key: keyof T) {
+        console.log(obj[key]);
+      }
+    `,
+    `
+      function getProperty<T, K extends keyof T>(obj: T, key: K) {
+        return obj[key];
+      }
+    `,
+    `
+      function box<T>(val: T) {
+        return { val };
+      }
+    `,
+    `
+      function doStuff<K, V>(map: Map<K, V>, key: K) {
+        let v = map.get(key);
+        v = 1;
+        map.set(key, v);
+        return v;
+      }
+    `,
+    `
+      function makeMap<K, V>() {
+        return new Map<K, V>();
+      }
+    `,
+    `
+      function makeMap<K, V>(ks: K[], vs: V[]) {
+        const r = new Map<K, V>();
+        ks.forEach((k, i) => {
+          r.set(k, vs[i]);
+        });
+        return r;
+      }
+    `,
+    `
+      function arrayOfPairs<T>() {
+        return [] as [T, T][];
+      }
+    `,
+    `
+      function isNonNull<T>(v: T): v is Exclude<T, null> {
+        return v !== null;
+      }
+    `,
+    `
+      function both<Args extends unknown[]>(
+        fn1: (...args: Args) => void,
+        fn2: (...args: Args) => void,
+      ): (...args: Args) => void {
+        return function (...args: Args) {
+          fn1(...args);
+          fn2(...args);
+        };
+      }
+    `,
+    `
+      function getLength<T extends { lenght: number }>(x: T) {
+        return x;
```

Comment:
I think this (and the next one) should be `x.length`?

In which case these both are invalid since they can be written:

```ts
function getLength(x: Lengthy) {
  return x.length;
}
```

If you want to keep them as is, rename them to `identity`.